src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -279,15 +279,14 @@ static void uct_tcp_sockcm_ep_mark_tx_completed(uct_tcp_sockcm_ep_t *cep)
     }
 }
 
-/**
- * This function should be called with the lock held.
- */
 ucs_status_t uct_tcp_sockcm_ep_progress_send(uct_tcp_sockcm_ep_t *cep)
 {
+    uct_tcp_sockcm_t UCS_V_UNUSED *tcp_sockcm = uct_tcp_sockcm_ep_get_cm(cep);
     ucs_status_t status;
     size_t sent_length;
     ucs_event_set_types_t events;
 
+    ucs_assert(ucs_async_is_blocked(tcp_sockcm->super.iface.worker->async));
```

Comment:
can we use UCP_WORKER_THREAD_CS_CHECK_IS_BLOCKED here?

Reply:
we can't use UCP macro in UCT code